scripts/release_main.js

```diff
@@ -0,0 +1,130 @@
+const path = require('path');
+
+const semver = require('semver');
+
+const {
+  askForConfirmation,
+  askForText,
+  checkGitStatus,
+  checkNpmAuth,
+  runExec,
+  revertChanges,
+  revertCommit,
+} = require('./release.helpers');
+
+const packageJsonPath = path.resolve(__dirname, '../package.json');
+const packageJson = require(packageJsonPath);
+
+async function checkGitBranchAndStatus() {
+  console.log('Checking your main branch...');
+  const resultBranch = await runExec('git branch --show-current', {
+    silent: true,
+  });
+  const branchName = resultBranch.stdout.toString().trim();
+  if (branchName !== 'main') {
+    console.error(
+      `🟠 You are at "${branchName}" instead of "main" branch. Are you sure you wanna release a stable version here?`
+    );
+    process.exit(1);
+  }
+
+  // Check if local main is up to date.
+  await runExec('git remote update', { silent: true });
+  const resultStatus = await runExec('git status -uno', { silent: true });
+  const mainStatus = resultStatus.stdout.toString().trim();
+
+  if (!mainStatus.includes("Your branch is up to date with 'origin/main'.")) {
+    console.error(`🟠 Please make sure your branch is up to date with the git repo.`);
+    process.exit(1);
+  }
+  await checkGitStatus();
+}
+
+function getNewVersion() {
+  const currentVersion = packageJson.version;
+
+  // TODO later - get the right version based on commit history.
+  const versionType = process.argv.slice(2)[0];
+  const versionBase = semver.coerce(currentVersion);
+  return semver.inc(versionBase, versionType) + '-beta.0';
```

Comment:
`beta.0` is maybe a little weird to me but I'm trying to get my thoughts correct here:

- We want to keep it unstable until we're ready for a correct `1.0.0`.
- So then wouldn't we be bumping the `beta.X` instead? and its `1.0.0.beta.123` until we're happy? 
- Thats not necessarily semver though... but for all patches, it should maybe do that instead?

Reply:
Honestly I'm not sure either, so here's my answer:

`beta.0` -> it's mandatory to have a number after any prefix. it cannot be just `.beta`

- My idea would be to release `0.1.0-beta.0` and then `0.x.x-beta.0` as we iterate. I'd prefer that over `.beta-x` so it's clear when new releases are features or fixes. Those this make sense?
- For that reason, I don't see us bumping `beta.x`. AFAIK `beta.x` is used to fix bugs directly related to that given version, not to fix bugs or release new feature in general.

Let me try to give an example. We start with `0.1.0-beta.0`.
- `0.2.0-beta.0` -> we now support this "foo1" thing.
- `0.2.0-beta.0` -> we now support this "bar" thing.
- `0.2.1-beta.0` -> we now fixed a bug at "biz" thing.

See, each commit it's about different features within the library as a whole. If we used `0.1.0-beta.x`, then it would be like:
- `0.2.0-beta.0` -> we now support this "foo1" thing.
- `0.2.0-beta.1` -> we now support this "foo2" thing.
- `0.2.0-beta.2` -> we now fixed a bug at "bar" thing.

It's not clear if the new versions are fixes, features, etc. 

In my understanding `beta` is used in the following scenario:
- `0.2.0-beta.0` -> we now support this "foo" thing.
- `0.2.0-beta.1` -> we fixed a bug related to the "foo" thing support this "foo2" thing.
- `0.2.0-beta.2` -> we fixed another bug related to the "foo" thing.
- `0.2.0-beta.3` -> we need to ship "bar" this. Does it make sense to be here? it should be "0.3.0-beta.0" instead.

Does this make sense? Honestly I don't know, I didn't find any document super clear about this, and looking at the NPM ecosystem around didn't bring me clarity either :/

Reply:
> beta.0 -> it's mandatory to have a number after any prefix. it cannot be just .beta

TIL 💡 

My idea would be to release 0.1.0-beta.0 and then 0.x.x-beta.0 as we iterate. I'd prefer that over .beta-x so it's clear when new releases are features or fixes. Those this make sense?
For that reason, I don't see us bumping beta.x. AFAIK beta.x is used to fix bugs directly related to that given version, not to fix bugs or release new feature in general.

Thats perfectly fine then, lets roll with it!

Reply:
Cool! Let me just rewrite `main` to add the co-authors, then rebase #2, and finally come back to this one. Give me ~10min.